File: app/src/main/java/com/kakacat/minitool/appInfo/presenter/AppInfoPresenter.java
package com.kakacat.minitool.appInfo.presenter;

import com.kakacat.minitool.appInfo.contract.AppInfoContract;
import com.kakacat.minitool.appInfo.model.AppInfoModel;
import com.kakacat.minitool.appInfo.model.bean.ApiPercentBean;
import com.kakacat.minitool.appInfo.model.bean.AppInfoBean;
import com.kakacat.minitool.common.util.ThreadUtil;

import java.util.List;

public class AppInfoPresenter implements AppInfoContract.Presenter {

    private AppInfoContract.View view;
    private AppInfoModel appInfoModel;

    public AppInfoPresenter(AppInfoContract.View view) {
        this.view = view;
        this.appInfoModel = AppInfoModel.getModelInstance();
    }

    @Override
    public void initData() {
        ThreadUtil.callInBackground(() -> {
            appInfoModel.initData(view.getContext().getPackageManager());
            ThreadUtil.callInUiThread(() -> view.onUpdateDataSet());
        });
    }

    @Override
    public void sortAppInfoList(int sortFlag) {
        ThreadUtil.callInBackground(() -> {
            appInfoModel.sortAppInfoList(sortFlag);
            ThreadUtil.callInUiThread(() -> view.onUpdateDataSet());
        });
    }

    @Override
    public List<ApiPercentBean> getApiPercentBeanList() {
        return appInfoModel.getApiPercentBeanList();
    }

    @Override
    public List<AppInfoBean> getAppInfoBeanList() {
        return appInfoModel.getAppInfoBeanList();
    }

}

File: app/src/main/java/com/kakacat/minitool/appInfo/presenter/AppInfoPresenter.kt
package com.kakacat.minitool.appInfo.presenter

import com.kakacat.minitool.appInfo.contract.AppInfoContract
import com.kakacat.minitool.appInfo.model.AppInfoModel
import com.kakacat.minitool.appInfo.model.bean.ApiPercentBean
import com.kakacat.minitool.appInfo.model.bean.AppInfoBean
import com.kakacat.minitool.common.util.SystemUtil
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.GlobalScope
import kotlinx.coroutines.async
import kotlinx.coroutines.launch

class AppInfoPresenter(private val view: AppInfoContract.View) : AppInfoContract.Presenter {

    private val appInfoModel by lazy { AppInfoModel() }
    
    override fun initData() {
        GlobalScope.launch {
            val job = GlobalScope.async { appInfoModel.initData(view.context.packageManager) }
            job.await()
            view.onUpdateDataSet()
        }
    }

    override fun sortAppInfoList(sortFlag: Int) {
        GlobalScope.launch {
            appInfoModel.sortAppInfoList(sortFlag)
            view.onUpdateDataSet()
        }
    }

    override val apiPercentBeanList: List<ApiPercentBean>
        get() = appInfoModel.getApiPercentBeanList()

    override val appInfoBeanList: List<AppInfoBean>
        get() = appInfoModel.getAppInfoBeanList()
}